Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt

```diff
@@ -10,34 +10,53 @@ import androidx.navigation.fragment.findNavController
 import de.rki.coronawarnapp.R
 import de.rki.coronawarnapp.databinding.FragmentOnboardingBinding
 import de.rki.coronawarnapp.ui.doNavigate
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.observe2
 import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import javax.inject.Inject
 
 /**
  * Onboarding starting point.
  */
-class OnboardingFragment : Fragment(R.layout.fragment_onboarding) {
+class OnboardingFragment : Fragment(R.layout.fragment_onboarding), AutoInject {
 
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val vm: OnboardingFragmentViewModel by cwaViewModels { viewModelFactory }
     private val binding: FragmentOnboardingBinding by viewBindingLazy()
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        binding.onboardingButtonNext.setOnClickListener {
-            findNavController().doNavigate(
-                OnboardingFragmentDirections.actionOnboardingFragmentToOnboardingPrivacyFragment()
-            )
+        binding.onboardingFragmentViewModel = vm
+        vm.routeToScreen.observe2(this) {
+            when (it) {
+                is OnboardingNavigationEvents.NavigateToOnboardingPrivacy ->
+                    navigateToOnboardingPrivacyFragment()
+                is OnboardingNavigationEvents.NavigateToEasyLanguageUrl ->
+                    navigateToEasyLanguageUrl()
+            }
         }
-        setLinks()
+        setButtonOnClickListener()
     }
 
-    private fun setLinks() {
-        binding.onboardingInclude.onboardingEasyLanguage
-            .setOnClickListener {
-                val browserIntent = Intent(
-                    Intent.ACTION_VIEW,
-                    Uri.parse(getString(R.string.onboarding_tracing_easy_language_explanation_url))
-                )
-                startActivity(browserIntent)
-            }
+    private fun setButtonOnClickListener() {
+        binding.onboardingButtonNext.setOnClickListener { vm.onNextButtonClick() }
+        binding.onboardingInclude.onboardingEasyLanguage.setOnClickListener { vm.onEasyLanguageClick() }
+    }
+
+    private fun navigateToOnboardingPrivacyFragment() {
+        findNavController().doNavigate(
+            OnboardingFragmentDirections.actionOnboardingFragmentToOnboardingPrivacyFragment()
+        )
+    }
+
+    private fun navigateToEasyLanguageUrl() {
```

Comment:
What do you think about also inlining this again? It's short and sweet, you can maybe use the extension `Fragment.doNavigate` to make it even shorter. :smile: